src/service/QuestionService.js

```diff
@@ -0,0 +1,104 @@
+import {
+    getPokemonById,
+    //getTypeById
+} from "../api/pokemons";
+
+
+// example result of calling getPokemonById(1):
+// { id: 1, name: 'Bulbasaur', photoUrl: 'someurl', types: [ { id: 3, name: 'grass' }, { id: 6, name: 'poison' } ] }
+// example result of calling getTypeById(3):
+// { id: 3, name: 'grass' }
+
+
+// example of getNextQuestion('guessName')
+// { 
+//     question: { pokemonName: 'Bulbasaur', pokemonUrl: 'someUrl', }, 
+//     answers: [ 'answer1', 'answer2', 'answer3', 'answer4', 'answer5-forTypes', 'answer6-forTypes', ], 
+//     correctAnswer: 'Bulbasaur' 
+// }
+
+export class QuestionService {
+
+    constructor(mode) {
+        this.mode = mode;
+    }
+
+    //Random nr 1-151
+    randomPokemonId() {
+        const firstGenPokemonLimit = 152
+        let randomId = Math.floor(Math.random() * firstGenPokemonLimit)
+        return randomId;
+    }
+
+    answers(numOfAnswers = 4) {
+        let answersArr = [];
+        for (let i = 1; i <= numOfAnswers; i++) {
+            answersArr.push(this.randomPokemonId());
+        }
+        return answersArr
+    }
+
+    correctAnswer(numOfAnswers = 4) {
+        return Math.floor(Math.random() * numOfAnswers) + 1;
+    }
+
+    getNextQuestion() {
+        const answersList = this.answers();
+        // console.log("List of ID answers " + answersList);
+        // console.log("Correct answer nr: " + this.correctAnswer());
+        const correctAnswerId = answersList[this.correctAnswer() - 1];
+        // console.log("Correct answer ID value " + correctAnswerId);
+
+        // Created to see what will be print
+        getPokemonById(correctAnswerId).then(meta => {
+            const correctAnswerPoke = meta;
+            console.log(correctAnswerPoke.name);
+        });
+
+        // There is a problem. I don't know how to reslove this promise
+        answersList.map( async (id) => {
```

Comment:
You should use something like `Promise.all()`

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all

but yeah, it is harder concept that simple one Promise :) we can pair up on it